### PR TITLE
ISPN-8874 Document multimap remote interface and protocol

### PR DESCRIPTION
**Pull request by karesti**
https://issues.jboss.org/browse/ISPN-8874

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/hotrod_protocol.adoc [hunk 1]
```diff
@@ -2034,13 +2034,26 @@ Here are the current values for each flag: +
...
 * Rollback, rollbacks a prepared transaction.
 
+. New operations to manipulate multimap caches.
+
+* request 0x67 - response 0x68 = Get a key and retrieve multiple values
+* request 0x69 - response 0x6A = Get a key with its metadata and retrieve multiple values
+* request 0x6B - response 0x6C = Put a key/value entry in the cache
+* request 0x6D - response 0x6E = Remove a key
+* request 0x6F - response 0x70 = Remove an entry
+* request 0x71 - response 0x72 = Get cache size
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/hotrod_protocol.adoc:2052]: @karesti Is "Get cache size" correct here? Or is it the size of a key/entry?

File: documentation/src/main/asciidoc/topics/ref_hotrod_multimapcachemanager.adoc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[id='remotecache_multimap_cachemanager-{context}']
+= Multimap Cache Manager
+The `MultimapCacheManager` interface is the entry point to get a `RemoteMultimapCache`.
+
+Hot Rod clients can perform multimap cache operations as in the following example:
+
+[source,java]
+----
+include::code_examples/MultimapCacheManagerRemote.java[]
+----
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_hotrod_multimapcachemanager.adoc:10]: Can we add a link to the Javadocs? I can only see `org.infinispan.multimap.api.embedded.MultimapCacheManager` when I build the Javadocs from the branch. Is there a remote API or is it the same as embedded?

**Review by oraNod — COMMENTED** [review]
Added a commit with a few suggestions as well as a couple of questions in here.

File: documentation/src/main/asciidoc/topics/hotrod_protocol.adoc [hunk 3]
```diff
@@ -2034,13 +2034,26 @@ Here are the current values for each flag: +
...
-It includes 3 new operations:
+This Hot Rod protocol version adds support for transaction operations and multimap caches.
+
+.New operations for transactions:
 
 * Prepare, with the transaction write set (i.e. modified keys), it tries to prepare and validate the transaction in the server.
 * Commit, commits a prepared transaction.
 * Rollback, rollbacks a prepared transaction.
 
+. New operations to manipulate multimap caches.
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/hotrod_protocol.adoc:2045]: any reason why this is in the middle of the transactions section?

File: documentation/src/main/asciidoc/topics/hotrod_protocol.adoc (hunk 1 above)
Comment (karesti) [reply documentation/src/main/asciidoc/topics/hotrod_protocol.adoc:2052]: size of the cache

File: documentation/src/main/asciidoc/topics/hotrod_protocol.adoc (hunk 3 above)
Comment (karesti) [reply documentation/src/main/asciidoc/topics/hotrod_protocol.adoc]: I moved it above

File: documentation/src/main/asciidoc/topics/ref_hotrod_multimapcachemanager.adoc (hunk 2 above)
Comment (karesti) [reply documentation/src/main/asciidoc/topics/ref_hotrod_multimapcachemanager.adoc:10]: This is a remote api but I don't think we need any link to the javadocs. I did like the counters

**Comment by karesti** [comment]
@pruivo @oraNod I reported the changes

File: documentation/src/main/asciidoc/topics/hotrod_protocol.adoc [hunk 4]
```diff
@@ -2034,8 +2034,23 @@ Here are the current values for each flag: +
...
+This Hot Rod protocol version adds support for transaction operations and multimap caches.
+
+It includes 9 new operations to manipulate multimap caches:
+
+* request 0x67 - response 0x68 = Get a key and retrieve multiple values
+* request 0x69 - response 0x6A = Get a key with metadata and retrieve multiple values
+* request 0x6B - response 0x6C = Put a key/value
+* request 0x6D - response 0x6E = Remove a key
+* request 0x6F - response 0x70 = Remove an entry
+* request 0x71 - response 0x72 = Size
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/hotrod_protocol.adoc:2046]:
```suggestion
* request 0x71 - response 0x72 = Get the cache size
```

**Review by oraNod — APPROVED** [review]
LGTM

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**